// react-qrious/lib/cjs.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var React = require('react');
var React__default = _interopDefault(React);
var QRious$1 = _interopDefault(require('qrious'));

/*! *****************************************************************************
Copyright (c) Microsoft Corporation. All rights reserved.
Licensed under the Apache License, Version 2.0 (the "License"); you may not use
this file except in compliance with the License. You may obtain a copy of the
License at http://www.apache.org/licenses/LICENSE-2.0

THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
MERCHANTABLITY OR NON-INFRINGEMENT.

See the Apache Version 2.0 License for specific language governing permissions
and limitations under the License.
***************************************************************************** */

var __assign = function() {
    __assign = Object.assign || function __assign(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};

function __rest(s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
}

var useQrious = function (options) {
    var qrious = React.useState(function () { return new QRious$1(options); })[0];
    var _a = React.useState(function () { return qrious.toDataURL(options.mime); }), dataUrl = _a[0], setDataUrl = _a[1];
    React.useEffect(function () {
        qrious.set(options);
        setDataUrl(qrious.toDataURL(options.mime));
    }, [options, qrious]);
    return [dataUrl, qrious];
};

var QRious = function (_a) {
    var background = _a.background, backgroundAlpha = _a.backgroundAlpha, foreground = _a.foreground, foregroundAlpha = _a.foregroundAlpha, level = _a.level, mime = _a.mime, padding = _a.padding, value = _a.value, size = _a.size, props = __rest(_a, ["background", "backgroundAlpha", "foreground", "foregroundAlpha", "level", "mime", "padding", "value", "size"]);
    var dataUrl = useQrious({
        background: background,
        backgroundAlpha: backgroundAlpha,
        foreground: foreground,
        foregroundAlpha: foregroundAlpha,
        level: level,
        mime: mime,
        padding: padding,
        size: size,
        value: value,
    })[0];
    return React__default.createElement('img', __assign(__assign({}, props), { src: dataUrl }));
};

exports.QRious = QRious;
exports.useQrious = useQrious;
